### PR TITLE
Introduce GitHub Actions PR Check job, fixes JBIDE-27703

### PR DESCRIPTION
**Pull request by odockal**

### CODE REVIEW

**Comment by odockal** [comment]
@sbouchet review please.

File: .github/workflows/build_test.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        path: ~/.m2/repository
+        key: jbosstools-${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+        restore-keys: |
+          jbosstools-${{ runner.os }}-maven-
+
+    # Build and compile using Maven
+    - name: Build/Compile and run unit tests
+      uses: GabrielBB/xvfb-action@v1
+      with:
+        run: mvn clean verify -U -fae --settings build-ci/maven-settings.xml -DskipITests=true -Dmaven.test.error.ignore=true -Dmaven.test.failure.ignore=true --quiet
```
Comment (sbouchet) [inline .github/workflows/build_test.yml:51]: please add --no-transfer-progress

**Comment by odockal** [comment]
@sbouchet review please.

File: .github/workflows/build_test.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      matrix:
+        os: [ubuntu-latest]
+        java: ["11"]
+      fail-fast: false
+
+    steps:
+    - name: Checkout PR branch
+      uses: actions/checkout@v2
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
```
Comment (sbouchet) [inline .github/workflows/build_test.yml:21]: please add fetch-depth: 0 to avoid baseline errors

File: .github/workflows/build_test.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+          jbosstools-${{ runner.os }}-maven-
+
+    # Build and compile using Maven
+    - name: Build/Compile and run unit tests
+      uses: GabrielBB/xvfb-action@v1
+      with:
+        run: >
+          mvn clean verify -U -fae --settings build-ci/maven-settings.xml 
+          -DskipITests=true -Dmaven.test.error.ignore=true 
+          -Dmaven.test.failure.ignore=true --quiet --no-transfer-progress
```
Comment (sbouchet) [inline .github/workflows/build_test.yml:54]: remove --quiet to better read logs.

**Review by sbouchet — CHANGES_REQUESTED** [review]

**Review by sbouchet — APPROVED** [review]
LGTM

**Pull request merged**